cassandra/src/main/java/io/zipkin/dependencies/spark/cassandra/ZipkinDependenciesJob.java

```diff
@@ -0,0 +1,172 @@
+import scala.runtime.AbstractFunction1;
+import zipkin.Codec;
+import zipkin.DependencyLink;
+import zipkin.Span;
+import zipkin.internal.Dependencies;
+import zipkin.internal.DependencyLinkSpan;
+import zipkin.internal.DependencyLinker;
+
+import static com.datastax.spark.connector.japi.CassandraJavaUtil.javaFunctions;
+import static zipkin.internal.Util.checkNotNull;
+import static zipkin.internal.Util.midnightUTC;
+
+public final class ZipkinDependenciesJob {
+  /** Runs with defaults, starting today */
+  public static void main(String[] args) {
+    new ZipkinDependenciesJob(new Builder()).run();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+
+    String keyspace = getEnv("CASSANDRA_KEYSPACE", "zipkin");
+
+    Map<String, String> sparkProperties = ImmutableMap.of(
+        "spark.ui.enabled", "false",
+        "spark.cassandra.connection.host", getEnv("CASSANDRA_HOST", "127.0.0.1"),
+        "spark.cassandra.connection.port", getEnv("CASSANDRA_PORT", "9042"),
+        "spark.cassandra.auth.username", getEnv("CASSANDRA_USERNAME", ""),
+        "spark.cassandra.auth.password", getEnv("CASSANDRA_PASSWORD", "")
+    );
+
+    // local[*] master lets us run & test the job locally without setting a Spark cluster
+    String sparkMaster = getEnv("SPARK_MASTER", "local[*]");
+
+    // By default the job only works on traces whose first timestamp is today
+    long day = midnightUTC(System.currentTimeMillis());
+
+    /** Keyspace to store dependency rowsToLinks. Defaults to "zipkin" */
+    public Builder keyspace(String keyspace) {
+      this.keyspace = checkNotNull(keyspace, "keyspace");
+      return this;
+    }
+
+    /** Day (in epoch milliseconds) to process dependencies for. Defaults to today. */
+    public Builder day(long day) {
+      this.day = midnightUTC(day);
+      return this;
+    }
+
+    public ZipkinDependenciesJob build() {
+      return new ZipkinDependenciesJob(this);
+    }
+
+    Builder() {
+    }
+  }
+
+  private static String getEnv(String key, String defaultValue) {
+    String result = System.getenv(key);
+    return result != null ? result : defaultValue;
+  }
+
+  final String keyspace;
+  final long day;
+  final String dateStamp;
+  final SparkConf conf;
+
+  ZipkinDependenciesJob(Builder builder) {
+    this.keyspace = builder.keyspace;
+    this.day = builder.day;
+    this.dateStamp = new SimpleDateFormat("yyyy-MM-dd").format(new Date(builder.day));
+    this.conf = new SparkConf(true)
+        .setMaster(builder.sparkMaster)
+        .setAppName(getClass().getName());
+    for (Map.Entry<String, String> entry : builder.sparkProperties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+  }
+
+  public void run() {
+    long microsLower = day * 1000;
+    long microsUpper = (day * 1000) + TimeUnit.DAYS.toMicros(1) - 1;
+
+    System.out.printf("Running Dependencies job for %s: %s ≤ Span.timestamp %s%n", dateStamp,
```

Comment:
ported things like System.out statements exactly as they were. Future pull requests will tidy up the code in general.